virttest/utils_test/__init__.py

```diff
@@ -755,11 +755,16 @@ def extract(vm, remote_path, dest_dir):
         extract_cmd = "tar %s %s -C %s" % (options,
                                            basename,
                                            os.path.dirname(dest_dir))
-        output = session.cmd(extract_cmd, timeout=120)
+        output = ""
+        while True:
+            output = session.cmd(extract_cmd, timeout=120)
+            if output != "":
+                break
```

Comment:
IMHO, There's no obvious improvement than sleep().
Even this method is worse, the loop maybe endless.
session.cmd() return means the extract command had been executed and finished. If the output is not your expectation, You should investigate why and find solution.  

If it's impossible to find the true reason and there's no better way to fix it, but by sleep().  I'm OK , but please add explanation for it.

I think you should investigate why